### ISSUE TITLE
Version 2.2.1 throws typing error when used with numba

### ISSUE THREAD

**Issue by c-leber** [ACCarnall/SpectRes#10]
Here is the error:
```
TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Unknown attribute 'warn' of type Module(<module 'warnings' from '/usr/local/lib/python3.9/warnings.py'>)

File "../../root/venv/lib/python3.9/site-packages/spectres/spectral_resampling_numba.py", line 126:
def spnumba(new_wavs, spec_wavs, spec_fluxes, spec_errs=None, fill=0,
    <source elided>
            if (j == 0 or j == new_wavs.shape[0]-1) and verbose:
                warnings.warn(
                ^

During: typing of get attribute at /root/venv/lib/python3.9/site-packages/spectres/spectral_resampling_numba.py (126)

File "../../root/venv/lib/python3.9/site-packages/spectres/spectral_resampling_numba.py", line 126:
def spnumba(new_wavs, spec_wavs, spec_fluxes, spec_errs=None, fill=0,
    <source elided>
            if (j == 0 or j == new_wavs.shape[0]-1) and verbose:
                warnings.warn(
                ^
```

Not an issue if I rollback to v2.2.0

**Comment by ACCarnall** [comment]
@WillJRoper will it affect you if I just revert this change for the Numba version of the code?

**Comment by WillJRoper** [comment]
It wouldn't affect us no. I avoid `numba` at all costs exactly for reasons like this 😅 (and its proclivity to return garbage without erroring).  I didn't realise they hadn't implemented a system for warnings in a `nopython` function though! There's an open issue about `numpy` warnings on their GitHub which appears to have died.

If you wanted to maintain symmetry with the non-numba code you could try using the `objmode` context manager, i.e:

``` python
from numba import objmode
if <warn_condition>:
    with objmode():
        warnings.warn(...)
```

But I have no idea if this would affect the performance of the jited code. Effectively (I think) anything wrapped with `objmode` will rejoin the python interpreter. In theory, it might have 0 penalty since it'll only rejoin if and when the warning is needed.

**Comment by WillJRoper** [comment]
Related: do you want some simple test workflows added to catch anything like this? A workflow testing that both flavours of spectres execute as expected would catch this sort of thing.

**Comment by ACCarnall** [comment]
Ok I've reverted the changes to the numba version and released v2.2.2, hopefully this fixes the problem. 

My personal view is that this is an extremely, extremely simple bit of code that really shouldn't need changing very often, if ever (and I like it that way). It's kind of you to offer to put more time into this, but I would really hope that this wouldn't be necessary going forwards.

Apologies for the disruption Christopher, very interesting though that you're (presumably) using this for problems way outside of astrophysics - I'd be very interested to hear more about what you're working on with it!
